server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/aeternity/rpc/AeternityRPCClient.java

```diff
@@ -0,0 +1,1087 @@
+        api = RestProxyFactory.createProxy(AeternityV2API.class, "https://roma-net.mdw.aepps.com");// aeternity official block explorer api, https://explorer.aepps.com
+        serviceConf = ServiceConfiguration.configure().baseUrl(rpcUrl).compile();
+    	accountService = new AccountServiceFactory().getService(serviceConf);
+    	chainService = new ChainServiceFactory().getService(serviceConf);
+    	transactionService = new TransactionServiceFactory().getService(TransactionServiceConfiguration.configure().baseUrl(rpcUrl).network(mainnet).compile());
+    }
+    
+	public BigDecimal getAccountBalance(String address) {
+    	BigDecimal returnBalance = BigDecimal.ZERO;
+    	try {
+			Account account = accountService.getAccount(address).blockingGet();
+			
+			BigInteger balance = account.getBalance();
+			if (balance.signum() == 1) {
+				returnBalance = new BigDecimal(balance);
+				returnBalance = returnBalance.divide(divider);
+	    	}
+    	}catch (Exception e) {
+    		if (e.getMessage().contains("Not Found")) return BigDecimal.ZERO;
+            log.error("Error reading balance.", e);
+            return null;
+		}
+    	return returnBalance;
+    }
+    
+	public List<String> getAddressTxIds(String address) throws GenericRpcException {
+		List<String> txIds = new ArrayList<String>();
+		Object txidsObject;
+		try {
+			txidsObject = api.getAccountTXs(address);
+			List<JSONObject> txidsObjectJSONarray = AeternityRPCClient.convertToJsonObjectArray(txidsObject);
+			for (int i=0; i< txidsObjectJSONarray.size(); i++) {
+				JSONObject txidObjectJSON = txidsObjectJSONarray.get(i);
+				JSONObject tx = txidObjectJSON.getJSONObject("tx");
+				if (tx.getString("type").equals("SpendTx") && tx.getString("recipient_id").equals(address)) {
+					txIds.add(txidObjectJSON.getString("hash"));
+				}
+			}
+		} catch (Exception e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+			throw new GenericRpcException(e);
+		}
+		return txIds;
+	}
+	
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	@Override
+	public NetworkInfo getNetworkInfo() throws GenericRpcException {
+		NetworkInfo networkInfo = null;
+		Object statusObject;
+		try {
+			statusObject = api.getStatus();
+			JSONObject statusObjectJSON = AeternityRPCClient.convertToJsonObject(statusObject);
+			JSONObject rootJsonObject = new JSONObject();
+			rootJsonObject.put("connections", statusObjectJSON.getInt("peer_count"));
+			HashMap<String,Object> result = new HashMap<String, Object>();
+			result = new ObjectMapper().readValue(rootJsonObject.toString(), HashMap.class);
+			networkInfo = new NetworkInfoWrapper((Map) result);
+			
+		} catch (Exception e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+			throw new GenericRpcException(e);
+		}
+		
+
+		return networkInfo;
+	}
+
+
+
+	@Override
+	public String getNewAddress() throws GenericRpcException {
+		try {
+		AeternityWallet newWallet = new AeternityWallet(null);
```

Comment:
???

Reply:
There is code in payment provider where new address is generated. In order to generate new address for Aeternity we generate new wallet.

Reply:
Remove saving wallet use seed/mnemonic and generate new addresses in deterministic fashion from there. Like it is done with XPUB.